src/lib/rehype/remarkInferToc.ts

```diff
@@ -0,0 +1,91 @@
+import type { Plugin } from "unified"
+import type { BlockContent, DefinitionContent, ListItem } from "mdast"
+import { toc, type Options } from "mdast-util-toc"
+import type { Nodes } from "mdast-util-toc/lib"
+import { visit } from "unist-util-visit"
+
+interface ITocNodeEntry {
```

Comment:
should be defined in `interfaces.ts`?

Reply:
Will pull all the types out of this file 👍